networkit/cpp/io/EdgeListReader.cpp

```diff
@@ -29,11 +29,20 @@ const std::map<std::string, node> &EdgeListReader::getNodeMap() const {
     return this->mapNodeIds;
 }
 
+struct pairhash {
+public:
+    template <typename T>
+    std::size_t operator()(const std::pair<T, T> &x) const {
+        return std::hash<T>()(x.first) ^ std::hash<T>()(313 * x.second);
+    }
+};
+
 Graph EdgeListReader::read(const std::string &path) {
     this->mapNodeIds.clear();
     MemoryMappedFile mmfile(path);
     auto it = mmfile.cbegin();
     auto end = mmfile.cend();
+    std::unordered_set<std::pair<node, node>, pairhash> insertedEdges;
```

Comment:
Instead of managing an additional set of edges, better as @michitux suggested, insert the edges without checks. After everything is inserted, sort each list (with either the already existing functionality or what @angriman introduced in #734) and remove duplicates. Also see here for comparison between hash-sets and vector-sort-unique: https://stackoverflow.com/questions/1041620/whats-the-most-efficient-way-to-erase-duplicates-and-sort-a-vector